statter.go

```diff
@@ -1,20 +1,420 @@
 package statter
 
 import (
-	"io"
+	"math"
+	"strconv"
+	"sync"
+	"sync/atomic"
 	"time"
+
+	"github.com/hamba/statter/internal/stats"
 )
 
-// Statter represents a stats instance.
-type Statter interface {
-	io.Closer
+// NullReporter is a reporter that discards all stats.
+var NullReporter = nullReporter{}
```

Comment:
This is inconsistent with other places where you called such implementations `discardXXX`